code/game/objects/structures/ladders.dm

```diff
@@ -13,20 +13,32 @@
 	desc = "An extremely sturdy metal ladder."
 
 
-/obj/structure/ladder/New()
-	spawn(8)
-		for(var/obj/structure/ladder/L in world)
-			if(L.id == id)
-				if(L.height == (height - 1))
-					down = L
-					continue
-				if(L.height == (height + 1))
-					up = L
-					continue
-
-			if(up && down)	//if both our connections are filled
-				break
-		update_icon()
+/obj/structure/ladder/Initialize(mapload)
+	if(!initialized)
+		ladders += src
+		..()

```

Comment:
Initialize with `mapload` set to TRUE checks it's return value. If it's FALSE or null nothing happens. If it evaluates to TRUE its added to a list in SSatoms. Once all atoms are mapload initialized SSatoms iterates this list and calls Initialize again.

Ladders need all ladders in the ladder list before calling `update_link()`. So what happens is this
- Map loaded
- First initialization pass on all atoms (mapload = TRUE)
- Ladders aren't initialized, so add self to ladder list and call parent (sets initialized to TRUE)
- Since mapload is TRUE it returns TRUE
- Since it returns TRUE SSatoms adds it to the late load list
- Finish first initialization pass
- Begin late initialization pass
- Initialize called again on ladders (mapload = FALSE)
- initialized was already set to TRUE by parent call in first pass, so that block is skipped
- all ladders now call `update_link()`